### PR TITLE
[DEV-10128] Add countries to search/spending_by_geography endpoint

### PR DESCRIPTION
**Pull request by aguest-kc**
**Description:**
Add a new `geo_layer` filter value of `country` that returns award and subaward spending by country.

**Technical details:**
* Update the Transactions ES index to add a `recipient_location_country_agg_key` field.
* Updated the `search/spending_by_geography` endpoint to use the `recipient_location_country_agg_key` or `pop_country_agg_key` to group ES transactions by country and return that amount for any/all countries included in the request.
* Updated the `search/spending_by_geography` endpoint to query SubawardSearch for all subawards where `sub_place_of_perform_country_co` or `sub_legal_entity_country_code` contains the country code included in the API request.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [DEV-123](https://federal-spending-transparency.atlassian.net/browse/DEV-10128):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected (API | Script | Download)
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
4. The nature of this work does not impact materialized views
5. The nature of this work does not require an impact assessment from the frontend
```

### CODE REVIEW

File: usaspending_api/search/v2/views/spending_by_geography.py [hunk 1]
```diff
@@ -397,6 +402,59 @@ def district_results(self, state_lookup: str, geo_queryset: QuerySet) -> List[di
...
+            country_queryset = country_queryset.filter(**{f"{loc_lookup}__in": self.geo_layer_filters})
+
+            ref_countries = RefCountryCode.objects.filter(country_code__in=self.geo_layer_filters).values(
+                "country_code", "country_name"
+            )
+            ref_countries = {country["country_code"]: country["country_name"] for country in ref_countries}
+        # If no specific countries were provided, then get all subawards grouped by country
+        else:
+            ref_countries = RefCountryCode.objects.all().values("country_code", "country_name")
+            ref_countries = {country["country_code"]: country["country_name"] for country in ref_countries}
```
Comment (collinwr) [inline usaspending_api/search/v2/views/spending_by_geography.py:427]: Not a big deal, but if line 427 matches 423 exactly, it should be brought out of the `if/else` block

**Review by collinwr — CHANGES_REQUESTED** [review]
Overall looks good! Looking forward to seeing this tested in QAT! Just left one comment/request

File: usaspending_api/search/v2/views/spending_by_geography.py (hunk 1 above)
Comment (aguest-kc) [reply usaspending_api/search/v2/views/spending_by_geography.py:427]: I didn't notice that I duplicated that line. I'll add a commit that pulls it out of the `if/else` block.

Comment (aguest-kc) [reply usaspending_api/search/v2/views/spending_by_geography.py]: Cleaned up the duplicate line in commit d11dd90

**Review by collinwr — APPROVED** [review]

**Pull request merged**